config/routes.rb

```diff
@@ -1,3 +1,10 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  devise_for :users, path: '', path_names: {sign_in: "signin", sign_out: "signout", sign_up: "signup"}
```

Comment:
- [ ] **[OPTIONAL]** A redundant message `You need to sign in or sign up before continuing.` is showing once I landed the sign up page or when I sign out. This message should only show when the user tries to go a page that he/she can't see. This behavior because we have default root which to posts which redirect us back to sign_in page. We can get rid of this message by using `authenticated` devise helper in routes, that way we have different roots for authenticated and unauthenticated users.
```
authenticated :user do
  root 'posts#index', as: :authenticated_user
end

devise_scope :user do
  root "devise/sessions#new"
end
```
then we can remove the root that you specified.

Reply:
fixed, thank you